pydantic/utils.py

```diff
@@ -465,19 +422,30 @@ def __init__(self, value: Any, items: Union['AbstractSetIntStr', 'MappingIntStrA
             self._items: Union['AbstractSetIntStr', 'MappingIntStrAny']
             self._type: Type[Union[set, dict]]  # type: ignore
 
-        # For further type checks speed-up
-        if isinstance(items, Mapping):
-            self._type = dict
-        elif isinstance(items, AbstractSet):
-            self._type = set
-        else:
-            raise TypeError(f'Unexpected type of exclude value {items.__class__}')
+        items = self._coerce_items(items)
+        self._type = dict
 
         if isinstance(value, (list, tuple)):
             items = self._normalize_indexes(items, len(value))
 
         self._items = items
 
+    @classmethod
+    def _coerce_items(cls, items: Union['AbstractSetIntStr', 'MappingIntStrAny']) -> 'MappingIntStrAny':
+        if isinstance(items, AbstractSet):
+            items = {k: ... for k in items}
```

Comment:
`dict.fromkeys()` does exact same thing and seems to be faster
```suggestion
            items = dict.fromkeys(items, ...)
```
```py
In [1]: items = {str(i) for i in range(100)}

In [2]: %timeit {k: ... for k in items}
6.39 µs ± 50.3 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [3]: %timeit dict.fromkeys(items, ...)
3.64 µs ± 31.4 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

Reply:
did this